src/Compilers/Core/Portable/DiagnosticAnalyzer/DefaultAnalyzerAssemblyLoader.Desktop.cs

```diff
@@ -33,15 +33,11 @@ internal class DefaultAnalyzerAssemblyLoader : AnalyzerAssemblyLoader
 
         private readonly Dictionary<AssemblyIdentity, Assembly> _loadedAssembliesByIdentity = new();
         private readonly Dictionary<string, AssemblyIdentity?> _loadedAssemblyIdentitiesByPath = new();
-
-        private int _hookedAssemblyResolve;
+        private bool _hookedAssemblyResolve;
 
         protected override Assembly LoadFromPathUncheckedImpl(string fullPath)
         {
-            if (Interlocked.CompareExchange(ref _hookedAssemblyResolve, 0, 1) == 0)
-            {
-                AppDomain.CurrentDomain.AssemblyResolve += CurrentDomain_AssemblyResolve;
-            }
+            EnsureResolvedHooked();
```

Comment:
Does this result in any behavior change?

Reply:
The ability to toggle the hook is necessary to determine if an assembly is in the _LoadFrom_ context. If we leave the hook enabled then the call to `Load(string name)` comes through the resolver, we find it in the directories, and then return the loaded assembly. Hence have to be able to turn the hook on / off during testing to probe _LoadFrom_ correctly.